_data-prepper/pipelines/configuration/processors/translate.md

```diff
@@ -0,0 +1,171 @@
+| :--- | :---  | :--- | :--- |
+| mappings | No | List | Defines inline mappings. For more information, see [mappings](#mappings). |
+| file | No | Map | Points to the file that contains mapping configurations. For more information, see [file](#file). |
+
+### mappings
+
+Each item in the `mappings` configuration contains the following options:
+
+| Parameter | Required | Type | Description |
+| :--- | :--- | :--- | :--- |
+| source | Yes | String or List | The source field to translate. Can be a string or a list of strings. |
+| targets | Yes | List | The target fields configurations, which can include the target field key, translation maps, and more. |
+
+Each item in the `targets` configuration contains the following options:
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| target | Yes | String | The key that specifies the field in the output where the translated value will be placed. |
+| map | No | Map | A list of key-value pairs that define the translations. Each key represents a possible value in the source field, and the corresponding value represents what it should be translated to. For examples, see [map option](#map-option). At least one of `map` and `regex` should be configured. |
+| regex | No | Map | A map of keys which defines the translation map. For more options, see [regex option](#regex-option). At least one of `map` and `regex` should be configured. |
+| default | No | String | The default value to use when no match is found during translation. |
+| type | No | String | Specify the type of the data for the target value |
+| translate_when | No | String | Use a [Data Prepper expression]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/) to specify a condition for performing the translation. If specified, the expression when translate only when the expression evaluates to `true`. |
+
+#### map option
+You can use the following key types when using the map option
+* Individual keys
+  ```yaml
+    map:
+      ok : "Success"
+      120: "Found"
+  ```
+* Number ranges
+  ```yaml
+    map:
+      "100-200": "Success"
+      "400-499": "Error"
+  ```
+* Comma-delimited keys
+  ```yaml
+    map:
+      "key1,key2,key3": "value1"
+      "100-200,key4": "value2"
+  ```
+
+While configuring the keys in map, do not use any overlapping number ranges or duplicate keys.
+
+#### regex option
+You can use the following options with the `regex` parameter. 
+
+| Parameter | Required | Type | Description |
+| :--- | :---  | :--- | :--- |
+| patterns | Yes | Map | A map of key-value pairs defining the regex patterns of keys and the value to translate to for each pattern. |
+| exact | No | Boolean | Whether to use full string match or partial string match on the regex patterns. If `true`, the pattern is considered a match only when the entire key matches the pattern. Otherwise, the pattern is considered a match as long as a sub-string of the key matches the pattern. |
```

Comment:
```suggestion
| exact | No | Boolean | Whether to use full string match or partial string match on the regex pattern. If `true`, the pattern is considered a match only when the entire key matches the pattern. Otherwise, the pattern is considered a match when a sub-string of the key matches the pattern. |
```